src/utils/market.util.ts

```diff
@@ -0,0 +1,140 @@
+import { MarketCoinsSortFieldEnum } from '../enums/market-coins-sort-field.enum';
+import { MarketCoin, MarketCoinRaw } from '../store/market/market.interfaces';
+import { Colors } from '../styles/colors';
+import { coingeckoApi, templeWalletApi } from './../api.service';
+
+export const fetchMarketTopCoins = () =>
+  coingeckoApi
+    .get<Array<MarketCoinRaw>>(
+      'coins/markets?vs_currency=usd&category=tezos-ecosystem&order=market_cap_desc&per_page=100&page=1&sparkline=false&price_change_percentage=24h%2C7d'
+    )
+    .then(({ data }) => {
+      const mappedCoins: Array<MarketCoin> = [];
+
+      for (const coinInfo of data) {
+        mappedCoins.push({
+          id: coinInfo.id,
+          name: coinInfo.name,
+          symbol: coinInfo.symbol.toUpperCase(),
+          imageUrl: coinInfo.image,
+          price: coinInfo.current_price,
+          priceChange7d: coinInfo?.price_change_percentage_7d_in_currency,
+          priceChange24h: coinInfo?.price_change_percentage_24h_in_currency,
+          volume24h: coinInfo.total_volume,
+          supply: coinInfo.circulating_supply,
+          marketCup: coinInfo.market_cap
+        });
+      }
+
+      return mappedCoins;
+    });
```

Comment:
no need in the creation of variable and use for loop here,  
`.then(response => response.data.map(coinInfo => ({ ... })))` will look much smaller